backend/src/main/java/fk/prof/backend/model/policy/impl/ZKWithCacheBasedPolicyStore.java

```diff
@@ -0,0 +1,175 @@
+package fk.prof.backend.model.policy.impl;
+
+import com.google.common.io.BaseEncoding;
+import com.google.protobuf.InvalidProtocolBufferException;
+import fk.prof.backend.model.policy.PolicyStore;
+import fk.prof.backend.proto.BackendDTO;
+import fk.prof.backend.util.ZookeeperUtil;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.curator.framework.CuratorFramework;
+import policy.PolicyDetails;
+import recording.Recorder;
+
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Zookeeper based implementation of the policy store
+ * Created by rohit.patiyal on 07/03/17.
+ */
+public class ZKWithCacheBasedPolicyStore implements PolicyStore {
```

Comment:
Added 'WithCache' in the name to make the use of cache map explicit but definitely can take it out of its name, will change that.